### PR TITLE
Programatic views - Part 2

### PR DESCRIPTION
**Pull request by kstoletniy**
#### Description:

* Implement all programatic views left: Home, SignIn and SignUp screens
* Delete Main.storyboard from the project

---

#### Tasks:
https://github.com/rootstrap/ios-base/issues/177
---

#### Risk:

* N/A

---

#### Preview:

https://user-images.githubusercontent.com/79856619/149201757-7168b498-8446-4ddd-9ecc-747bfb899a2a.mov

### CODE REVIEW

File: ios-base/Extensions/TextFieldExtension.swift [hunk 1]
```diff
@@ -9,6 +9,27 @@
 import UIKit
 
 extension UITextField {
+  static func primaryTextField(
```
Comment (glm4) [inline ios-base/Extensions/TextFieldExtension.swift:12]: I think we should make this an initializer instead, it sounds a bit repetitive and we currently only have primary so it doesn't mean much.

File: ios-base.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -166,6 +163,7 @@
 		F9BE994CFA78189FABF22B93 /* Pods-ios-base-ios-baseUITests.release.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-ios-base-ios-baseUITests.release.xcconfig"; path = "Target Support Files/Pods-ios-base-ios-baseUITests/Pods-ios-base-ios-baseUITests.release.xcconfig"; sourceTree = "<group>"; };
 		FA54D5871E11C59600F0DBEA /* FirstViewController.swift */ = {isa = PBXFileReference; fileEncoding = 4; indentWidth = 2; lastKnownFileType = sourcecode.swift; path = FirstViewController.swift; sourceTree = "<group>"; tabWidth = 2; };
 		FABDC9211EE1EB2B000DDAC3 /* ConfigurationManager.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = ConfigurationManager.swift; sourceTree = "<group>"; };
+		FD66EC19270F708100BF65B3 /* GoogleService-Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; name = "GoogleService-Info.plist"; path = "ios-base/Resources/GoogleService-Info.plist"; sourceTree = SOURCE_ROOT; };
```
Comment (glm4) [inline ios-base.xcodeproj/project.pbxproj:166]: I think you should merge master into your branch and get the latests for this file

File: ios-base/Home/Routes/HomeRoutes.swift [hunk 3]
```diff
@@ -15,9 +15,7 @@ enum HomeRoutes: Route {
   var screen: UIViewController {
     switch self {
     case .home:
-      guard let home = R.storyboard.main.homeViewController() else {
-        return UIViewController()
-      }
+      let home = HomeViewController()
       home.viewModel = HomeViewModel()
```
Comment (glm4) [inline ios-base/Home/Routes/HomeRoutes.swift:19]: This is an opportunity to inject the viewModel and make it a private constant.

File: ios-base/Home/Views/HomeViewController.swift [hunk 4]
```diff
@@ -23,26 +51,64 @@ class HomeViewController: UIViewController, ActivityIndicatorPresenter {
...
+      deleteAccountButton.bottomAnchor.constraint(
+        equalTo: view.bottomAnchor,
+        constant: -UI.Defaults.margin
+      ),
+      logOutButton.bottomAnchor.constraint(
+        equalTo: deleteAccountButton.topAnchor,
+        constant: -UI.Button.spacing
+      )
+    ])
+
```
Comment (glm4) [inline ios-base/Home/Views/HomeViewController.swift:107]: pls remove extra line

File: ios-base/Info.plist [hunk 5]
```diff
@@ -36,7 +36,7 @@
 	<key>CFBundleVersion</key>
 	<string>1</string>
 	<key>ConfigurationName</key>
-	<string>Debug</string>
+	<string>Staging</string>
```
Comment (glm4) [inline ios-base/Info.plist:39]: pls don't commit this change

File: ios-base/Onboarding/Routes/OnboardingRoutes.swift [hunk 6]
```diff
@@ -26,17 +26,13 @@ enum OnboardingRoutes: Route {
...
     signIn.viewModel = SignInViewModelWithCredentials()
     return signIn
   }
 
   private func buildSignUpViewController() -> UIViewController {
-    guard let signUp = R.storyboard.main.signUpViewController() else {
-      return UIViewController()
-    }
+    let signUp = SignUpViewController()
     signUp.viewModel = SignUpViewModelWithEmail()
```
Comment (glm4) [inline ios-base/Onboarding/Routes/OnboardingRoutes.swift:36]: same here, we should now inject the viewmodels into the view controllers via the initalizer

File: ios-base/Onboarding/Views/SignInViewController.swift [hunk 7]
```diff
@@ -47,13 +64,49 @@ class SignInViewController: UIViewController, ActivityIndicatorPresenter {
...
+      logInButton
+    ])
+    
+    activateConstrains()
+  }
+  
+  func activateConstrains() {
+    [titleLabel, emailField, passwordField, logInButton].forEach({
+      $0.attachHorizontally(to: view)
+    })
```
Comment (glm4) [inline ios-base/Onboarding/Views/SignInViewController.swift:94]:
```suggestion
    [titleLabel, emailField, passwordField, logInButton].forEach {
      $0.attachHorizontally(to: view)
    }
```

File: ios-base/Onboarding/Views/SignUpViewController.swift [hunk 8]
```diff
@@ -50,13 +74,61 @@ class SignUpViewController: UIViewController, ActivityIndicatorPresenter {
...
+  func activateConstrains() {
+    [
+      titleLabel,
+     emailField,
+     passwordField,
+     passwordConfirmationField,
+     signUpButton
+    ].forEach({
+      $0.attachHorizontally(to: view)
+    })
```
Comment (glm4) [inline ios-base/Onboarding/Views/SignUpViewController.swift:112]:
```suggestion
    ].forEach {
      $0.attachHorizontally(to: view)
    }
```

File: ios-base/Onboarding/Views/SignUpViewController.swift [hunk 9]
```diff
@@ -50,13 +74,61 @@ class SignUpViewController: UIViewController, ActivityIndicatorPresenter {
...
+    activateConstrains()
+  }
+  
+  func activateConstrains() {
+    [
+      titleLabel,
+     emailField,
+     passwordField,
+     passwordConfirmationField,
+     signUpButton
```
Comment (glm4) [inline ios-base/Onboarding/Views/SignUpViewController.swift:109]: check indentation

**Review by LetoFranco — APPROVED** [review]

**Review by germanStabile — APPROVED** [review]
Looks good!

**Review by LetoFranco — APPROVED** [review]

**Review by glm4 — APPROVED** [review]

**Pull request merged**